src/main/java/com/teragrep/jos_01/procfs/status/Vmstat.java

```diff
@@ -0,0 +1,234 @@
+        nr_anon_transparent_hugepages,
+        nr_free_cma,
+        nr_dirty_threshold,
+        nr_dirty_background_threshold,
+        pgpgin,
+        pgpgout,
+        pswpin,
+        pswpout,
+        pgalloc_dma,
+        pgalloc_dma32,
+        pgalloc_normal,
+        pgalloc_high,
+        pgalloc_movable,
+        pgfree,
+        pgactivate,
+        pgdeactivate,
+        pgfault,
+        pgmajfault,
+        pgrefill_dma,
+        pgrefill_dma32,
+        pgrefill_normal,
+        pgrefill_high,
+        pgrefill_movable,
+        pgsteal_kswapd_dma,
+        pgsteal_kswapd_dma32,
+        pgsteal_kswapd_normal,
+        pgsteal_kswapd_high,
+        pgsteal_kswapd_movable,
+        pgsteal_direct_dma,
+        pgsteal_direct_dma32,
+        pgsteal_direct_normal,
+        pgsteal_direct_high,
+        pgsteal_direct_movable,
+        pgscan_kswapd_dma,
+        pgscan_kswapd_dma32,
+        pgscan_kswapd_normal,
+        pgscan_kswapd_high,
+        pgscan_kswapd_movable,
+        pgscan_direct_dma,
+        pgscan_direct_dma32,
+        pgscan_direct_normal,
+        pgscan_direct_high,
+        pgscan_direct_movable,
+        pgscan_direct_throttle,
+        zone_reclaim_failed,
+        pginodesteal,
+        slabs_scanned,
+        kswapd_inodesteal,
+        kswapd_low_wmark_hit_quickly,
+        kswapd_high_wmark_hit_quickly,
+        pageoutrun,
+        allocstall,
+        pgrotated,
+        drop_pagecache,
+        drop_slab,
+        numa_pte_updates,
+        numa_huge_pte_updates,
+        numa_hint_faults,
+        numa_hint_faults_local,
+        numa_pages_migrated,
+        pgmigrate_success,
+        pgmigrate_fail,
+        compact_migrate_scanned,
+        compact_free_scanned,
+        compact_isolated,
+        compact_stall,
+        compact_fail,
+        compact_success,
+        htlb_buddy_alloc_success,
+        htlb_buddy_alloc_fail,
+        unevictable_pgs_culled,
+        unevictable_pgs_scanned,
+        unevictable_pgs_rescued,
+        unevictable_pgs_mlocked,
+        unevictable_pgs_munlocked,
+        unevictable_pgs_cleared,
+        unevictable_pgs_stranded,
+        thp_fault_alloc,
+        thp_fault_fallback,
+        thp_collapse_alloc,
+        thp_collapse_alloc_failed,
+        thp_split,
+        thp_zero_page_alloc,
+        thp_zero_page_alloc_failed,
+        balloon_inflate,
+        balloon_deflate,
+        balloon_migrate,
+        nr_tlb_remote_flush,
+        nr_tlb_remote_flush_received,
+        nr_tlb_local_flush_all,
+        nr_tlb_local_flush_one,
+        vmacache_find_calls,
+        vmacache_find_hits,
+        vmacache_full_flushes
+    };
+
+    public Vmstat(ArrayList<String> rows) {
+        this.rows = rows;
+        statistics = new LinkedHashMap<String, String>();
+        Pattern pattern = Pattern.compile("(.*) (.*)");
+        for (String row : rows) {
+            Matcher matcher = pattern.matcher(row);
+            if (matcher.find()) {
```

Comment:
See the first `matcher.find()` comment

Reply:
This one is a newline-separated file and has a redundant for-loop! Removed.